src/test/scala/chiselTests/RecordSpec.scala

```diff
@@ -0,0 +1,45 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import scala.collection.immutable.ListMap
+
+class RecordSpec extends ChiselFlatSpec {
+  class RecordFromMap(val elements: ListMap[String, Data]) extends Record {
+    override def cloneType =
```

Comment:
I think this is a little nicer:
override def cloneType = (new RecordFromMap(
  for((field, elem) <- elements) yield field -> elem.cloneType
)).asInstanceOf[this.type]

or at least,

override def cloneType = (new RecordFromMap(
  elements.map(e => e._1 -> e._2.cloneType)
)).asInstanceOf[this.type]